File: Source/ipc/examples/java/Reader.java
import com.mittudev.ipc.Connection;
import com.mittudev.ipc.Message;
import com.mittudev.ipc.ConnectionCallback;
import java.lang.Thread;

class Callback implements ConnectionCallback{
	public int recived = 0;
	public void onMessage(Message msg){
		System.out.println(new String(msg.getData()));
		recived++;
	}
}

public class Reader{
	public static void main(String[] args) {
		Callback cb = new Callback();
		Connection conn = new Connection("ipcdemo", 1);
		conn.setCallback(cb);
		conn.startAutoDispatch();

		while(cb.recived < 3){
			try{
				Thread.sleep(100);
			}catch (Exception e){
				e.printStackTrace();
			}
		}

		conn.stopAutoDispatch();
		conn.close();
		conn.destroy();
	}
}

File: Source/ipc/examples/node/reader.js
var IPC = require('../../node/build/Release/IPC');

var recived = 0;

var conn = new IPC.Connection("ipcdemo", 1);

function waituntil(num, final){
  if (recived < num){
    setTimeout(function() { waituntil(num, final); }, 500);
  }else {
    final()
  }
}

function cleanup(){
  conn.stopAutoDispatch();
  conn.close();
}


conn.setCallback(msg => {
  console.log(msg.getData().toString());
  recived++;
});

conn.startAutoDispatch();

waituntil(3, cleanup);
